blockchain/blockchain.go

```diff
@@ -450,54 +450,66 @@ func (bc *BlockChain) loadLastState() error {
 	return nil
 }
 
-// SetHead rewinds the local chain to a new head. In the case of headers, everything
-// above the new head will be deleted and the new one set. In the case of blocks
-// though, the head may be further rewound if block bodies are missing (non-archive
-// nodes after a fast sync).
 func (bc *BlockChain) SetHead(head uint64) error {
-	logger.Info("Rewinding blockchain", "target", head)
+	logger.Warn("Rewinding blockchain", "target", head)
 
-	bc.mu.Lock()
-	defer bc.mu.Unlock()
+	bc.chainmu.Lock()
```

Comment:
mutex is changed, is this okay?

Reply:
Ethereum also changed mutex at https://github.com/ethereum/go-ethereum/commit/f25f776c9f9d6a0538cd9028d9d82fc6965d9123. Take a look.

Reply:
@hqjang-pepper Your reference combined two mutexes into `bc.chainmu`. But your suggestion changes only one usage of the `bc.mu` mutex. It means that `SetHead` is not a critical section of `bc.mu` mutex and other codes still using `bc.mu` can be executed concurrently with `SetHead`. 
I think this mutex change is not proper for the intend of this PR.

Reply:
@aidan-kwon I reflected. Thanks for your comment.